app/controllers/root_controller.rb

```diff
@@ -7,6 +7,26 @@ def index
   private
 
   def destination
-    current_user_or_visitor.is_logged_in? ? dashboard_path : new_user_session_path
+    if current_user_or_visitor.is_logged_in?
+      logged_in_destination
+    else
+      logged_out_destination
+    end
+  end
+
+  def logged_out_destination
+    new_user_session_path
+  end
+
+  def logged_in_destination
+    case current_user_groups.count
+    when 0 then explore_path
+    when 1 then current_user_groups.first
+    else        dashboard_path
+    end
+  end
+
+  def current_user_groups
+    @current_user_groups ||= current_user_or_visitor.groups
```

Comment:
Is this ever so slightly different logic to on the client side? What about the case where a user is a member of a parent group and its subgroup?